app/models/vha_facility_adapter.rb

```diff
@@ -1,69 +1,69 @@
 # frozen_string_literal: true
-require_dependency 'facilities/client'
-
-class VAHealthFacility < ActiveModelSerializers::Model
-  attr_accessor :station_id, :station_number, :visn_id, :name, :classification, :lat, :long,
-                :address, :phone, :hours, :services
+# frozen_string_literal: true
+class VHAFacilityAdapter
+  VHA_URL = +ENV['VHA_MAPSERVER_URL']
+  VHA_LAYER = ENV['VHA_MAPSERVER_LAYER']
+  VHA_ID_FIELD = 'StationNum'
+  FACILITY_TYPE = 'va_health_facility'
 
-  def self.query(bbox:, services:)
-    results = client.query(bbox: bbox.join(','),
-                           where: VAHealthFacility.where_clause(services))
-    results.each_with_object([]) do |record, facs|
-      facs << VAHealthFacility.from_gis(record)
-    end
+  def initialize
+    @client = Facilities::Client.new(url: VHA_URL, layer: VHA_LAYER, id_field: VHA_ID_FIELD)
   end
 
-  def self.find_by_id(id:)
-    results = client.get(identifier: id)
-    VAHealthFacility.from_gis(results.first) unless results.blank?
+  def query(bbox, services = nil)
+    @client.query(bbox: bbox.join(','),
+                  where: VHAFacilityAdapter.where_clause(services))
   end
 
-  def self.service_whitelist
-    SERVICE_HIERARCHY.flatten(2)
+  def find_by(id:)
+    @client.get(identifier: id)
   end
 
   def self.where_clause(services)
     services.map { |s| "#{s}='YES'" }.join(' AND ') unless services.nil?
   end
 
-  def self.from_gis(record)
+  def from_gis(record)
     attrs = record['attributes']
-    m = VAHealthFacility.from_gis_attrs(TOP_KEYMAP, attrs)
-    m[:address] = VAHealthFacility.from_gis_attrs(ADDR_KEYMAP, attrs)
-    m[:phone] = VAHealthFacility.from_gis_attrs(PHONE_KEYMAP, attrs)
-    m[:hours] = VAHealthFacility.from_gis_attrs(HOURS_KEYMAP, attrs)
-    m[:services] = VAHealthFacility.services_from_gis(attrs)
-    VAHealthFacility.new(m)
+    m = from_gis_attrs(TOP_KEYMAP, attrs)
+    m[:facility_type] = FACILITY_TYPE
+    m[:address] = {}
+    m[:address][:physical] = from_gis_attrs(ADDR_KEYMAP, attrs)
+    m[:address][:physical][:zip] = attrs['Zip']
+    m[:address][:physical][:zip] << '-' + attrs['Zip4'] unless attrs['Zip4'].strip.empty?
+    m[:address][:mailing] = {}
+    m[:phone] = from_gis_attrs(PHONE_KEYMAP, attrs)
+    m[:hours] = from_gis_attrs(HOURS_KEYMAP, attrs)
+    m[:services] = services_from_gis(attrs)
+    VAFacility.new(m)
+  end
```

Comment:
In the current GIS data set, empty values are represented as an empty string - I can pull down the complete facilities data and verify. But, it looks like it's possible for ArcGIS REST to return nulls. I think only the `Zip4` processing above would be thrown off by a nil, so I'll fix that up.